xarray/test/test_dataset.py

```diff
@@ -2494,3 +2494,45 @@ def test_setattr_raises(self):
             ds.foo = 2
         with self.assertRaisesRegexp(AttributeError, 'cannot set attr'):
             ds.other = 2
+
+    def test_get_variables_by_attributes(self):
```

Comment:
I am not very creative with the tests. Basically just example cases and not really unittests.  Let me know if that is OK.